compiler/src/main/kotlin/edu/cornell/cs/apl/viaduct/selection/SelectionConstraintGenerator.kt

```diff
@@ -62,17 +62,17 @@ class SelectionConstraintGenerator(
 
     // TODO: pc must be weak enough for the hosts involved in the selected protocols to read it
     fun viableProtocols(node: LetNode): Set<Protocol> =
-        protocolFactory.viableProtocols(node).filter {
+        node.protocol?.let { setOf(it.value) } ?: protocolFactory.viableProtocols(node).filter {
```

Comment:
Do we wanna check if the annotations are valid here?

Reply:
I don't think so. The protocol annotation should be respected regardless of what the protocol factory says. We should treat it like programmer intent.

Reply:
But protocol annotations should agree with what's actually implementable by the `CodeGenerators`. The lack of checks allows programmers to cause internal compiler errors and weird behavior.